test/stat_tracker_test.rb

```diff
@@ -57,7 +57,33 @@ def test_it_has_game_teams
   # GAME STATISTICS
 
   def test_highest_total_score
-    assert_equal 5, @stat_tracker.highest_total_score
+
+    #Here I avoid loading the csv data all together
+    #But I still have a real instance of StatTracker to work with
+    game_path = './fixtures/no_data_fixture.csv'
+    team_path = './fixtures/no_data_fixture.csv'
+    game_teams_path = './fixtures/no_data_fixture.csv'
+
+    locations = {
+      games: game_path,
+      teams: team_path,
+      game_teams: game_teams_path
+    }
+    stat_tracker = StatTracker.new(locations)
+    game_1 = mock('game_1')
+    game_2 = mock('game_2')
+    game_3 = mock('game_3')
+    games = [game_1, game_2, game_3]
+    games.each_with_index do |game, index|
```

Comment:
This method may be new to some of you. It tracks the index position of every element in the array as we iterate. So `game_1.total_goals` will have a stubbed value of `0`, `game_2.total_goals` will have a stubbed value of `1` and `game_3.total_goals` will have a stubbed value of `2`

Reply:
Let me see if I understand this:

- As you indexed it created a value for each mock.  Is that value now considered a "stub-value"?

So instead of writing
```ruby
game_2 = mock('game_2')
game_2.stubs(:total_goals).returns(1)
```
You automated the return value (which is an integer?) That now gets calculated as part of the "score" that would've excited in the CSV file.

Am I on the right track with this, or did I totally miss it?

Reply:
@Arique1104 Yes, you are correct in that assumption and yes the return value is an integer for each of my stubs in this context. And yes, my mocks stand in place for actual game objects that would need to be created via the csv data. Remember that you can always clone down a repo, switch to the branch some functionality lives on and then poke around.